packages/ui/src/utils/utilsFormat.ts

```diff
@@ -1,12 +1,20 @@
+import { detect, fromUrl, fromNavigator } from '@lingui/detect-locale'
 import BigNumber from 'bignumber.js'
 import isUndefined from 'lodash/isUndefined'
 import isNaN from 'lodash/isNaN'
 
-import { detectLocale } from 'ui/src/utils/utilsLocale'
-
 BigNumber.config({ EXPONENTIAL_AT: 20, ROUNDING_MODE: BigNumber.ROUND_HALF_UP })
 export const BN = BigNumber
 
+export function detectLocale() {
+  if (typeof window !== 'undefined') {
+    const DEFAULT_FALLBACK = () => 'en'
+    return detect(fromUrl('lang'), fromNavigator(), DEFAULT_FALLBACK) ?? 'en-US'
+  } else {
+    return 'en-US'
+  }
+}
+
```

Comment:
```suggestion
export const detectLocale = () =>
  (typeof window !== 'undefined' && detect(fromUrl('lang'), fromNavigator(), () => 'en')) ?? 'en-US'
```

Reply:
Decided to inline it directly as `const localeDetected =  (typeof window !== 'undefined' && detect(fromUrl('lang'), fromNavigator(), () => 'en')) || 'en-US'` given the function is literally used only once.